buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantCommandTask.groovy

```diff
@@ -32,19 +38,37 @@ import javax.inject.Inject
 public class VagrantCommandTask extends LoggedExec {
 
     @Input
+    String command
+
+    @Input @Optional
+    String subcommand
+
+    @Input
     String boxName
 
     @Input
     Map<String, String> environmentVars
 
     public VagrantCommandTask() {
         executable = 'vagrant'
-
         project.afterEvaluate {
-            // It'd be nice if --machine-readable were, well, nice
-            standardOutput = new TeeOutputStream(standardOutput, createLoggerOutputStream())
-            if (environmentVars != null) {
-                environment environmentVars
+            // Schedule an action to set the values for exec before it runs.
+            // If you don't do this from within 'afterEvaluate', then
+            // the exec and configure actions are run out of order.
+            doFirst {
```

Comment:
I think we should leave it as is for the sake of least astonishment. The biggest thing I struggled with in this PR was that configurations I wanted to preempt or override from the superclass were being captured before my subclass's logic could change them because of how `afterEvaluate` schedules things. In a sense, the superclass logic was happening before the subclasses logic could override it, which is something that I imagine most people wouldn't expect.